tests/integration/session_test.go

```diff
@@ -0,0 +1,33 @@
+// Copyright 2023 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package integration
+
+import (
+	"testing"
+
+	"code.gitea.io/gitea/models/auth"
+	"code.gitea.io/gitea/models/db"
+	"code.gitea.io/gitea/models/unittest"
+	"code.gitea.io/gitea/tests"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func Test_RegenerateSession(t *testing.T) {
+	defer tests.PrepareTestEnv(t)()
+
+	assert.NoError(t, unittest.PrepareTestDatabase())
+
+	exist, err := auth.ExistSession(db.DefaultContext, "new_key")
+	assert.NoError(t, err)
+	assert.False(t, exist)
+
+	sess, err := auth.RegenerateSession(db.DefaultContext, "", "new_key")
+	assert.NoError(t, err)
+	assert.EqualValues(t, "new_key", sess.Key)
+
+	sess, err = auth.ReadSession(db.DefaultContext, "new_key2")
+	assert.NoError(t, err)
+	assert.EqualValues(t, "new_key2", sess.Key)
```

Comment:
Here, we should also check the data and expiration date.

Reply:
Added data, but expire date cannot guarantee in the same second.